src/optimproblem.cpp

```diff
@@ -897,6 +1066,63 @@ void OptimProblem::rollOut(Vec x){
       }
     }
   }
+
+}
+
+void OptimProblem::updateLagrangian(const double prev_mu, const Vec x, Vec lambda) {
+
+  evalF(x, lambda, true);
```

Comment:
Why call evalF() if you ignore the return value? Does evalF() have any side effects?

Reply:
Updating Lagrangian needs the intermediate discontinuities. `evalF` is simply to store intermediate final states, so we don't need the objective function itself.